test/test_gridder_padarray.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+from fatiando import mesher,gridder
+from fatiando.gravmag import prism
+from numpy.testing import assert_array_almost_equal as assert_almost
+from nose.tools import assert_raises
+from nose.tools import assert_equal
+
+def test_fails_if_bad_pad_operation():
+    'gridder.pad_array fails if given a bad padding array operation option'
+    p = 'foo'
+    s = (100,100)
+    x,y,z = gridder.regular((-1000.,1000.,-1000.,1000.),s,z=-150)
+    g = z.reshape(s)
+    assert_raises(ValueError,gridder.pad_array,g,padtype=p)
+
+def test_pad_and_unpad_equal_2d():
+    'gridder.pad_array and subsequent .unpad_array gives original array: 2D'
+    s = (100,101)
+    x,y,z = gridder.regular((-5000.,5000.,-5000.,5000.),s,z=-150)
+    model = [mesher.Prism(-4000,-3000,-4000,-3000,0,2000,{'density':1000}),
+        mesher.Prism(-1000,1000,-1000,1000,0,2000,{'density':-900}),
+        mesher.Prism(2000,4000,3000,4000,0,2000,{'density':1300})]
+    gz = prism.gz(x,y,z,model)
+    gz = gz.reshape(s)
+    gpad,_,nps = gridder.pad_array(gz)
+    gunpad = gridder.unpad_array(gpad,nps)
+    assert_almost(gunpad,gz)
+
+def test_pad_and_unpad_equal_1d():
+    'gridder.pad_array and subsequent .unpad_array gives original array: 1D'
+    x = np.random.rand(21)
```

Comment:
It's better to use a defined random seed in unit tests to make sure that you can always repeat a failing test later on.